ui/src/router/components/form/components/router_config/GeneralSettingsPanel.js

```diff
@@ -60,6 +70,59 @@ export const GeneralSettingsPanel = ({
 
         <EuiSpacer size="m" />
 
+        <EuiFormRow
+          fullWidth
+          label={
+            <FormLabelWithToolTip
+              label="Protocol *"
+              content="Specify the type of Turing router Protocol"
+            />
+          }
+          display="row"
+        >
+          <EuiSuperSelect
+            fullWidth
+            options={protocolTypeOptions}
+            valueOfSelected={protocol}
+            onChange={(e) => {
+              onChange("config.protocol")(e);
+              // update service_method for each route if changed to UPI
+              onChange("config.routes")(
+                routes.map((route) => {
+                  return {
+                    ...route,
+                    service_method:
```

Comment:
I guess we are not showing the service method anywhere. Though it's auto-populated, I think it would be better to show it (as a non-editable field or by [appending](https://elastic.github.io/eui/#/forms/form-controls#prepend-and-append) to the endpoint, to make it clearer to the user. wdyt?